Content.Shared/Movement/Systems/SharedJetpackSystem.cs

```diff
@@ -20,182 +21,278 @@ public abstract class SharedJetpackSystem : EntitySystem
+            _physics.SetBodyStatus(user, physics, BodyStatus.OnGround);
 
-        _movementSpeedModifier.RefreshWeightlessModifiers(uid);
+        _movementSpeedModifier.RefreshWeightlessModifiers(user);
     }
 
-    private void OnJetpackToggle(EntityUid uid, JetpackComponent component, ToggleJetpackEvent args)
+    private void OnJetpackToggle(Entity<JetpackComponent> jetpack, ref ToggleJetpackEvent args)
     {
         if (args.Handled)
             return;
 
-        if (TryComp(uid, out TransformComponent? xform) && !CanEnableOnGrid(xform.GridUid))
-        {
-            _popup.PopupClient(Loc.GetString("jetpack-no-station"), uid, args.Performer);
+        var user = args.Performer;
+        var (uid, component) = jetpack;
+        var toggled = !component.Enabled;
 
+        // If they're already using a jetpack that isn't this one and trying to turn this one on, don't let them.
+        if (TryComp<JetpackUserComponent>(user, out var userComponent) && userComponent.Jetpack != uid)
+        {
+            _popup.PopupClient(Loc.GetString("jetpack-already-using"), user);
             return;
         }
 
-        SetEnabled(uid, component, !IsEnabled(uid));
+        // You can still turn the jetpack on/off when on a grid that doesn't permit flying; you just won't be able to fly!
+        var canFly = TryComp(uid, out TransformComponent? xform) && CanFlyOnGrid(xform.GridUid);
+        SetEnabled(jetpack, toggled, null, canFly);
+
+        args.Handled = true;
     }
 
-    private bool CanEnableOnGrid(EntityUid? gridUid)
+    /// <remarks>
+    /// This should return only whether you can <i>fly</i> with the jetpack, assuming it's turned on etc.
+    /// Return value should stay regardless of <see cref="CanEnable"/>.
+    /// </remarks>
+    private bool CanFlyOnGrid(EntityUid? gridUid)
     {
         // No and no again! Do not attempt to activate the jetpack on a grid with gravity disabled. You will not be the first or the last to try this.
         // https://discord.com/channels/310555209753690112/310555209753690112/1270067921682694234
-        return gridUid == null ||
-               (!HasComp<GravityComponent>(gridUid));
+        return gridUid == null || !HasComp<GravityComponent>(gridUid);
     }
 
     private void OnJetpackGetAction(EntityUid uid, JetpackComponent component, GetItemActionsEvent args)
     {
         args.AddAction(ref component.ToggleActionEntity, component.ToggleAction);
     }
 
-    private bool IsEnabled(EntityUid uid)
-    {
-        return HasComp<ActiveJetpackComponent>(uid);
-    }
-
-    public void SetEnabled(EntityUid uid, JetpackComponent component, bool enabled, EntityUid? user = null)
+    /// <summary>
+    /// Sets the provided jetpack to whether it is enabled, and if so, whether it can fly.
+    /// <paramref name="enabled"/> is only whether the jetpack should appear active, but
+    /// <paramref name="flyIfEnabled"/> is whether the user should actually be able to fly with it.
```

Comment:
I think your parameter descriptions are detailed enough that you can leave these two lines out. (Thanks for adding docs, but of course no good deed goes unpunished :^)).